service/turn.rest/src/main/java/org/kaazing/gateway/service/turn/rest/DefaultCredentialsGenerator.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+import java.time.Instant;
+import java.util.Base64;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.security.auth.Subject;
+
+/**
+ * Default implementation of the TurnRestCredentialsGenerator
+ * It contains a standard implementation of the short term credentials 
+ * as described in <a href='https://tools.ietf.org/html/draft-uberti-behave-turn-rest-00'>draft-uberti-behave-turn-rest-00</a> 
+ */
+public class DefaultCredentialsGenerator implements TurnRestCredentialsGenerator {
+
+    public static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
```

Comment:
Specifying the algorithm makes this a specific implementation of the [rest draft](https://tools.ietf.org/html/draft-uberti-behave-turn-rest-00'>draft-uberti-behave-turn-rest-00) instead of a generic one as the draft says any HMAC algorithm can be used.

Further research on Keytool importpass showed that the imported pass is a implementation of [SecretKey](https://docs.oracle.com/javase/7/docs/api/javax/crypto/SecretKey.html) and has an get Algorithm associated with that, in which HMAC_SHA1_ALGORITHM could be attached.

As such I would (and reviewed with JF) that the API for the service be 

```
<properties>
       <feature.turn.rest>true</feature.turn.rest>
       <generate.credentials>class:org.kaazing.gateway.service.turn.rest.DefaultCredentialsGeneratorImpl</generate.credentials>
       <key.alias>logen</key.aliasy>

       <options>
           <credentials.ttl>43200</credentials.ttl>
           <username.separator>:</username.separator>
       </options>
       <uris>
           <!-- these are only examples -->
           <uri>turn:coturn:3478?proto=udp</uri>
       </uris>
   </properties>
```

Where key alias is the alias to the SecretKey with algorithm stored in the truststore.  If the alias does not exist the gateway should fail to start.  (We only support secure setup of service).

With that you could then change the API of setSharedKey to setSharedSecret and use use the SharedSecret class directly.

From there the DefaultCredentialsGenerator can use the getAlogrithm method to see which type of MAC class it can instantiate